simple/dgram_waitset.c

```diff
@@ -84,6 +84,13 @@ static int send_recv()
 	struct fi_cq_entry comp;
 	int ret;
 
+	ret = fi_recv(ep, rx_buf, rx_size + ft_rx_prefix_size(), fi_mr_desc(mr),
```

Comment:
> Should this be an ft_post_rx()?

No, anytime something is posted with `ft_post_rx()` a global counter is updated indicating how many receives have been posted. Whenever `ft_rx()` is run (in `ft_sync()`), it tries to reap all completions that have been posted. So it tries to reap `total posted` - `current` completions. This leads to a hang since the poll timeout is configured to `-1`.

To me it seemed the easiest way to handle this was to just post directly with `fi_recv` and circumvent the polling.